docs/kerberos.md

```diff
@@ -0,0 +1,124 @@
+---
+post_title: Kerberos
+nav_title: Kerberos
+menu_order: 120
+enterprise: 'no'
+---
+
+
+# HDFS Kerberos
+
+Kerberos is an authentication system to allow Spark to retrieve and write data securely to a Kerberos-enabled HDFS cluster. As of Mesosphere Spark `2.2.0-2`, long-running jobs will renew their delegation tokens (authentication credentials). This section assumes you have previously set up a Kerberos-enabled HDFS cluster. For instruction on how to setup a secure HDFS cluster see the [HDFS documentation][http://example.com]
+
+## Spark Installation
+
+Spark (and all Kerberos-enabed) components need a valid `krb5.conf` file. You can setup the Spark service to use a single `krb5.conf` file for all of the it's drivers.
+
+1.  A `krb5.conf` file tells Spark how to connect to your KDC.  Base64 encode this file:
+
+        cat krb5.conf | base64 -w 0 
+
+1.  Put the encoded file (as a string) into your JSON configuration file:
+
+        {
+           "security": {
+             "kerberos": {
+              "krb5conf": "<base64 encoding>"
+              }
+           }
+        }
+        
+     Your configuration will probably also have the `hdfs` parameters from above:
+     
+        {
+          "service": {
+              "name": "kerberized-spark",
+              "user": "nobody"
+          },
+          "hdfs": {
+              "config-url": "http://api.hdfs.marathon.l4lb.thisdcos.directory/v1/endpoints"
+          },
+          "security": {
+              "kerberos": {
+                  "krb5conf": "<base64_encoding>"
+            }
+          }
+        }
+
+        
+1.  Install Spark with your custom configuration, here called `options.json`:
+
+        dcos package install --options=/path/to/options.json spark
+        
+1. If you've enabled the history server via `history-server.enabled`, you must also configure the principal and keytab for the history server.  **WARNING**: The keytab contains secrets, in the current history server package the keytab is not stored securely. See [Limitations][9]
+
+    Base64 encode your keytab:
+
+        cat spark.keytab | base64
+
+    And add the following to your configuration file:
+
+         {
+            "history-server": {
+                "kerberos": {
+                  "principal": "spark@REALM",
+                  "keytab": "<base64 encoding>"
+                }
+            }
+         }
+
+## Job Submission
+
+To authenticate to a Kerberos KDC, Spark on Mesos supports keytab files as well as ticket-granting tickets (TGTs). Keytabs are valid infinitely, while tickets can expire. Keytabs are recommended, especially for long-running streaming jobs.
+
+### Keytab Authentication
+
+Submit the job with the keytab:
+
+    dcos spark run --submit-args="\
+    --kerberos-principal user@REALM \
+    --keytab-secret-path /__dcos_base64__hdfs-keytab \
+    --conf ... --class MySparkJob <url> <args>"
```

Comment:
@ArtRand I don't know if we want to document it, but to get this working in Soak with CentOS, I had to add this:
`--conf spark.mesos.driverEnv.SPARK_USER=root`, to get the executors to run as `root`.

Reply:
All of the docs relating to deployment in strict and permissive modes is here (https://github.com/mesosphere/dcos-docs-enterprise/blob/master/service-docs/spark/spark-auth.md). I would be in favor of moving all of this to it's own section. WDYT?

Reply:
What are you proposing to move to its own section? The installing as `root` in CentOS? -- Since it only comes up for Kerberos, I was thinking it belonged in the `kerberos.md`.

Reply:
I'll add a note. But I think that installing as `root` on CentOS should be in the limitations or install notes. I'll add it there. Do you have a description of the problem?

Reply:
Yeah, I agree. We can copy the blurb from the Cassandra docs: https://docs.mesosphere.com/service-docs/cassandra/2.0.3-3.0.14/limitations/#service-user

Reply:
Do you know/remember the source of the problem that required you to set `--conf spark.mesos.driverEnv.SPARK_USER=root`? I'll add the note, but I'd like to have some context.